src/ValueConverters/PrependTypeConverter.cs

```diff
@@ -0,0 +1,50 @@
+﻿#region Copyright information
+// <copyright file="BLoc.cs">
+//     Licensed under Microsoft Public License (Ms-PL)
+//     https://github.com/XAMLMarkupExtensions/WPFLocalizationExtension/blob/master/LICENSE
+// </copyright>
+// <author>Konrad Mattheis</author>
+#endregion
+
+namespace WPFLocalizeExtension.ValueConverters
+{
+    #region Usings
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+    using System.Windows.Data;
+    using System.Windows.Markup;
+    #endregion
+
+    /// <summary>
+    /// PrependTypeConverter allows to prepend the type of the value as string with the default _ separator. To change the default separator just us the converterparamater
+    /// </summary>
+    public class PrependTypeConverter : MarkupExtension, IValueConverter
+    {
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            if (value != null)
+            {
+                var separator = "_";
+                if (parameter != null && parameter.GetType() == typeof(string))
+                    separator = parameter.ToString();
+                return value.GetType().Name + separator + value.ToString();
+            }
+
+            return null;
+        }
+
+        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override object ProvideValue(IServiceProvider serviceProvider)
+        {
+            return new PrependTypeConverter();
```

Comment:
I think here can be used "this". It will allow to create less objects. Also this code can be moved in separate class "BaseConverter" and used here and in TranslateConverter